hpat/distributed_analysis.py

```diff
@@ -306,6 +306,17 @@ def _analyze_call(self, lhs, func_var, args, array_dists):
             self._analyze_call_np_concatenate(lhs, args, array_dists)
             return
 
+        if call_list == ['permutation', 'random', np]:
+            call_def = guard(get_definition, self.func_ir, func_var)
+            if isinstance(call_def, ir.Expr) and call_def.op == 'getattr':
```

Comment:
call definition check is not required since the full call name is checked in call_list.